java-daemon-endpoints/src/main/java/net/cryptic_game/backend/endpoints/network/NetworkMemberEndpoints.java

```diff
@@ -1,10 +1,111 @@
 package net.cryptic_game.backend.endpoints.network;
 
-import net.cryptic_game.backend.base.api.endpoint.ApiEndpointCollection;
+import com.google.gson.JsonObject;
+import net.cryptic_game.backend.base.api.endpoint.*;
+import net.cryptic_game.backend.data.device.Device;
+import net.cryptic_game.backend.data.network.Network;
+import net.cryptic_game.backend.data.network.NetworkInvitation;
+import net.cryptic_game.backend.data.network.NetworkMember;
+import net.cryptic_game.backend.data.user.User;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
 
 public class NetworkMemberEndpoints extends ApiEndpointCollection {
 
     public NetworkMemberEndpoints() {
-        super("network/member");
+        super("network_member");
+    }
+
+    @ApiEndpoint("members")
+    public ApiResponse member(@ApiParameter("user_id") final UUID userId,
+                              @ApiParameter("device") final UUID deviceId) {
+        final User user = User.getById(userId);
+        final Device device = Device.getById(deviceId);
+
+        if (!device.isPoweredOn())
+            return new ApiResponse(ApiResponseType.FORBIDDEN, "DEVICE_NOT_ONLINE");
+
+        return new ApiResponse(ApiResponseType.OK, NetworkMember.getMembershipsOfDevice(device));
+
+    }
+
+    @ApiEndpoint("request")
+    public ApiResponse request(@ApiParameter("user_id") final UUID userId,
+                          @ApiParameter("device") final UUID deviceId,
+                          @ApiParameter("id") final UUID id) {
+        final User user = User.getById(userId);
+        final Device device = Device.getById(deviceId);
+        final Network network = Network.getById(id);
+
+        if (device == null || !device.hasUserAccess(user)) {
+            return new ApiResponse(ApiResponseType.FORBIDDEN, "DEVICE_ACCESS_DENIED");
+        }
+
+        if (!device.isPoweredOn()) {
+            return new ApiResponse(ApiResponseType.FORBIDDEN, "DEVICE_NOT_ONLINE");
+        }
+
+        if (network == null) {
+            return new ApiResponse(ApiResponseType.FORBIDDEN, "NETWORK_NOT_FOUND");
+        }
+
+        if (NetworkMember.getMember(network, device) != null) {
+            return new ApiResponse(ApiResponseType.FORBIDDEN, "ALREADY_MEMBER_OF_NETWORK");
+        }
+
+        if (NetworkInvitation.getInvitation(network, device) != null) {
+            return new ApiResponse(ApiResponseType.ALREADY_EXISTS, "INVITATION_ALREADY_EXISTS");
+        }
+        return new ApiResponse(ApiResponseType.OK, NetworkInvitation.createInvitation(network, device, null));
+    }
+
+    @ApiEndpoint("invitations")
+    public ApiResponse invitations(@ApiParameter("user_id") final UUID user_id,
+                              @ApiParameter("device") final UUID device_id,
+                              @ApiParameter("id") final UUID id){
```

Comment:
Make the parameter `device` and `id` optional. 
If none is given: return `BAD_REQUEST`
If only the device is given: return all NetworkInvitation's of that device.
If only the network is given: return all NetworkInvitation's of that network
If both, the network and the device is given: return the NetworkInvitation of that network to the device if existing, otherwise return `INVITATION_NOT_FOUND`
Don't forget to check if the user has the needed access permissions.

Reply:
what message should return if Device or id were null?

Reply:
`BAD_REQUEST` with message `MISSING_DEVICE_ID_OR_NETWORK_ID`